force-app/main/default/classes/MeetupTriggerHandler.cls

```diff
@@ -0,0 +1,43 @@
+/* 
+* @description Responsible for all Meetup sObject trigger behaviors.
+*/
+public with sharing class MeetupTriggerHandler {
+    public void onBeforeInsert(List<Meetup__c> meetupRecords) {
+        generateRegistrationCodes(meetupRecords);
+    }
+    
+    /* 
+	* @description Populates the given meetup records with a unique Registration Code.
+    * @param meetupRecords The list of meetup records to operate on.
+    */
+    public void generateRegistrationCodes(List<Meetup__c> meetupRecords) {
+        Schema.DescribeFieldResult registrationCodeField = Meetup__c.RegistrationCode__c.getDescribe();
+        Integer registrationCodeLength = registrationCodeField.getLength();
+        
+        // Known limitation, if we exceed 50k records then an async
+        // process will be required to ensure uniqueness of codes.
```

Comment:
There's probably a better approach for long term random unique code generation. Another option would be to attempt insert and have a retry pattern if the DML fails due to existing code.